util/types/datum.go

```diff
@@ -762,12 +762,16 @@ func (d *Datum) convertToString(target *FieldType) (Datum, error) {
 		return invalidConv(d, target.Tp)
 	}
 	// TODO: consider target.Charset/Collate
+	var err error
+	if target.Flen > 0 && target.Flen < len(s) {
+		err = ErrDataTooLong
```

Comment:
MySQL error only has column name and row number, which we don't have in this function.